src/blocks/posts/index.php

```diff
@@ -0,0 +1,417 @@
+	$list_items_class  = '';
+
+	if ( 'horizontal' !== $block_style ) {
+
+		$list_items_class .= 'flex-col';
+
+	}
+
+	foreach ( $posts as $post ) {
+
+		$list_class       = '';
+		$image_class      = '';
+		$align_self_class = '';
+
+		if ( isset( $attributes['listPosition'] ) && 'horizontal' === $block_style ) {
+
+			if ( 'left' === $attributes['listPosition'] ) {
+
+				$image_class .= ' mr-2 sm:mr-3';
+			} else {
+
+				$image_class     .= ' ml-3 sm:ml-3';
+				$list_items_class = 'flex-row-reverse';
+
+			}
+		}
+
+		if ( isset( $attributes['imageSize'] ) && 'horizontal' === $block_style ) {
+
+			$image_class .= ' ' . $attributes['imageSize'];
+
+		} else {
+
+			$image_class .= 'w-full relative';
+
+		}
+
+		$list_items_markup .= sprintf(
+			'<li class="flex flex-auto %1$s items-stretch w-full mt-2 mb-2 ml-0 pl-0">',
+			$list_items_class
+		);
+
+		if ( null !== $post['thumbnailURL'] && $post['thumbnailURL'] ) {
+
+			if ( 'stacked' === $block_style ) {
+
+				$image_class .= ' mb-2';
+
+			}
+
+			$list_items_markup .= sprintf(
+				'<div class="wp-block-coblocks-posts__image table flex-0 %1$s"><a href="%2$s" class="block w-full bg-cover bg-center-center pt-full" style="background-image:url(%3$s)"></a></div>',
+				esc_attr( $image_class ),
+				esc_url( $post['postLink'] ),
+				esc_url( $post['thumbnailURL'] )
+			);
+
+			if ( 'horizontal' === $block_style && ( isset( $attributes['displayPostContent'] ) && ! $attributes['displayPostContent'] ) && ( isset( $attributes['columns'] ) && 2 >= $attributes['columns'] ) ) {
+
+				$align_self_class = 'self-center';
+			}
+		} else {
+			$align_self_class = 'flex-start';
+		}
+
+		$list_items_markup .= sprintf(
+			'<div class="wp-block-coblocks-posts__content flex flex-col %s w-full">',
+			esc_attr( $align_self_class )
+		);
+
+		if ( isset( $attributes['displayPostDate'] ) && $attributes['displayPostDate'] && 'stacked' === $block_style ) {
+
+			$list_items_markup .= sprintf(
+				'<time datetime="%1$s" class="wp-block-coblocks-posts__date mb-1">%2$s</time>',
+				esc_url( $post['date'] ),
+				esc_html( $post['dateReadable'] )
+			);
+
+		}
+
+		$title = $post['title'];
+
+		if ( ! $title ) {
+
+			$title = _x( '(no title)', 'placeholder when a post has no title', 'coblocks' );
+
+		}
+
+		$list_items_markup .= sprintf(
+			'<a href="%1$s" alt="%2$s">%2$s</a>',
+			$post['postLink'],
+			$title
+		);
+
+		if ( isset( $attributes['displayPostDate'] ) && $attributes['displayPostDate'] && 'horizontal' === $block_style ) {
+
+			$list_items_markup .= sprintf(
+				'<time datetime="%1$s" class="wp-block-coblocks-posts__date mt-2">%2$s</time>',
+				esc_url( $post['date'] ),
+				esc_html( $post['dateReadable'] )
+			);
+
+		}
+
+		if ( isset( $attributes['displayPostContent'] ) && $attributes['displayPostContent'] ) {
+
+			$post_excerpt    = $post['postExcerpt'];
+			$trimmed_excerpt = esc_html( wp_trim_words( $post_excerpt, $attributes['excerptLength'], ' &hellip; ' ) );
```

Comment:
Do not trim words here. We should be receiving the excerpt through the use of [`get_the_excerpt`](https://developer.wordpress.org/reference/functions/get_the_excerpt/)